aggregator/src/aggregator/http_handlers.rs

```diff
@@ -166,8 +166,12 @@ where
     T: Encode + Sync + Send + 'static,
 {
     async fn run(&self, conn: Conn) -> Conn {
-        conn.with_header(KnownHeaderName::ContentType, self.media_type)
-            .ok(self.object.get_encoded())
+        match self.object.get_encoded() {
+            Ok(encoded) => conn
+                .with_header(KnownHeaderName::ContentType, self.media_type)
+                .ok(encoded),
+            Err(e) => Error::from(e).run(conn).await,
```

Comment:
Good catch. I added an `Error::ResponseEncode(prio::codec::CodecError)` variant which we explicitly instantiate here. So we still get `From<CodecError> for Error` to use most anywhere else. There's some other places where we can fail to handle a request because `get_encoded` fails. I'll see if it's practical to use `Error::ReponseEncode` instead of `Error::MessageDecode` for those.

Reply:
As I'd hoped, this just called for a handful of `.map_err(Error::ResponseEncode)` calls. I'm a little concerned by how we to just remember to opt out of the default, which is to wrap `CodecError` into `Error::MessageDecode` via `Into`.